### PR TITLE
[Merged by Bors] - feat: countable filtration in a countably generated measurable space

### PR DESCRIPTION
**Pull request by RemyDegenne**
In a countably generated measurable space `α`, we can build a sequence of finer and finer finite measurable partitions of the space such that the measurable space is generated by the union of all partitions.
This sequence of partitions (`countablePartition α n`) is defined in `MeasureTheory.MeasurableSpace.CountablyGenerated`. This is a new file in which we put the definition of `CountablyGenerated` (which was previously in MeasurableSpace.Basic).

In `Probability.Process.CountablyGenerated`, we build the filtration generated by `countablePartition α n` for all `n : ℕ`, which we call `countableFiltration α`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/MeasurableSpace/CountablyGenerated.lean [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+/-- We say a measurable space is countably generated
+if it can be generated by a countable set of sets. -/
+class CountablyGenerated (α : Type*) [m : MeasurableSpace α] : Prop where
+  isCountablyGenerated : ∃ b : Set (Set α), b.Countable ∧ m = generateFrom b
+#align measurable_space.countably_generated MeasurableSpace.CountablyGenerated
+
+/-- A countable set of sets that generate the measurable space.
+We insert `∅` to ensure it is nonempty. -/
+def generatingSet (α : Type*) [MeasurableSpace α] [h : CountablyGenerated α] : Set (Set α) :=
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/MeasurableSpace/CountablyGenerated.lean:50]: This name is a little bit too generic. `countableGeneratingSet` would probably be better, as the point is really the countability.

**Review by sgouezel — COMMENTED** [review]
I have an issue with the design here. You are starting from a countably generated measurable space, pick some countable generating set, and then construct very useful object from it. However, I have several examples in mind where I would like to do this using a specific countable generating set with nice properties, not the one that is provided by choice. And also examples where I would like to play this game, but the countable set would not generate the whole sigma-algebra.

Do you think it would be possible to develop a general framework starting from a structure `N -> Set \alpha` for which all sets are measurable, and associate to it all the objects in your construction? And then specialize  it to your countable generating set?

**Comment by sgouezel** [comment]
The PR still has the label `awaiting-author`. Are you done with your changes, or is there something else you want to do?

**Comment by RemyDegenne** [comment]
I did not finish the generalization of most lemmas to arbitrary sequences of sets yet (and I had an almost lean-free week). I'll update this tomorrow.

**Review by sgouezel — COMMENTED** [review]
LGTM, thanks!
bors d+

**Comment by RemyDegenne** [comment]
Thanks for the review!
bors r+

**Pull request closed without merging**